pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -356,10 +372,17 @@ private List<String> findCommonBrokerInstances(Set<String> brokerTenants) {
   }
 
   // return the union of brokerTenants from the tables list.
-  private Set<String> getBrokerTenantsUnion(List<TableConfig> tableConfigList) {
+  private Set<String> getBrokerTenantsUnion(@Nullable List<TableConfig> tableConfigList,
```

Comment:
Same comment as above, it doesn't seem like `tableConfigList` can be `null`, only empty (same as `logicalTableConfigList`) so it looks a little strange that we're handling the two differently.